mipengine/common/node_catalog.py

```diff
@@ -91,6 +86,19 @@ def get_nodes_with_datasets(self, datasets: List[str]) -> List[List[LocalNode]]:
         """
         return [self._nodes_per_dataset[dataset] for dataset in datasets]
 
+    def get_nodes_with_any_of_datasets(self, datasets: List[str]) -> List[LocalNode]:
```

Comment:
get_nodes_with_datasets could be completely removed and get_nodes_with_any_of_datasets refactored to something like:
```
    def get_nodes_with_any_of_datasets(self, datasets: List[str]) -> List[LocalNode]:
        nodes_per_datasets = [self._nodes_per_dataset[dataset] for dataset in datasets]
        nodes = []
        for nodes_per_dataset in nodes_per_datasets:
            for node in nodes_per_dataset:
                if node not in nodes:
                    nodes.append(node)

        if not nodes:
            raise Exception(f"There are no nodes with any of the datasets->{datasets}")

        return nodes
```